File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s)==0:
            return 0
        
        i,j,longestLength= 0,1,0
        while i<len(s):
            length = 1
            while j<len(s):
                if s[j] in s[i:j]:
                    break
                else:
                    length+=1
                j+=1
            if length>longestLength:
                longestLength = length
            i+=1
            j=i+1 
        return longestLength
        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int> mp;
        int i=0,j=0;
        int maxi = 0;
        while(j<s.size()){
            if(mp[s[j]]==1){
                mp[s[i]]--;
                i++;
            }else{
                mp[s[j]]++;
                j++;
            }
            maxi = max(maxi,j-i);
        }
        return maxi;
    }
};